src/classes/RD2_RecurringDonation.cls

```diff
@@ -71,7 +71,7 @@ public inherited sharing class RD2_RecurringDonation {
     * @param rd Recurring Donation record
     */
     public RD2_RecurringDonation(npe03__Recurring_Donation__c rd) {
-        this(rd, rd.npe03__Donations__r);
+        this(rd, rd?.npe03__Donations__r);
```

Comment:
add a null check, so RD2_RecurringDonation can be used without querying the schedule data